### PR TITLE
feat(cockpit): migrate legacy cockpit resources to v1 API

### PR DESCRIPTION
**Pull request by jremy42**
close https://github.com/scaleway/terraform-provider-scaleway/issues/2569

### CODE REVIEW

**Review by remyleone — DISMISSED** [review]

File: internal/services/cockpit/cockpit.go [hunk 1]
```diff
@@ -5,7 +5,7 @@ import (
 
 	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
-	cockpit "github.com/scaleway/scaleway-sdk-go/api/cockpit/v1beta1"
+	"github.com/scaleway/scaleway-sdk-go/api/cockpit/v1"
```
Comment (yfodil) [inline internal/services/cockpit/cockpit.go:8]: Timeouts are not necessary anymore

File: docs/data-sources/cockpit.md [hunk 2]
```diff
@@ -22,7 +25,7 @@ data "scaleway_cockpit" "main" {}
 ```hcl
 // Get a specific project's cockpit
 data "scaleway_cockpit" "main" {
-  project_id = "11111111-1111-1111-1111-111111111111"
+project_id = "11111111-1111-1111-1111-111111111111"
```
Comment (yfodil) [inline docs/data-sources/cockpit.md:28]: Please fix indentation

File: docs/data-sources/cockpit.md [hunk 3]
```diff
@@ -35,9 +38,9 @@ data "scaleway_cockpit" "main" {
...
-    - `metrics_url` - The metrics URL
-    - `logs_url` - The logs URL
-    - `alertmanager_url` - The alertmanager URL
-    - `grafana_url` - The grafana URL
+- `plan_id` - (Deprecated) The ID of the current plan
+- `endpoints` - (Deprecated) Endpoints
+- `metrics_url` - (Deprecated) The metrics URL
+- `logs_url` - (Deprecated) The logs URL
+- `alertmanager_url` - (Deprecated) The alertmanager URL
+- `grafana_url` - (Deprecated) The grafana URL
```
Comment (yfodil) [inline docs/data-sources/cockpit.md:46]: Please fix indentation

File: docs/resources/cockpit.md [hunk 4]
```diff
@@ -15,17 +15,26 @@ For more information consult the [documentation](https://www.scaleway.com/en/doc
...
 resource "scaleway_cockpit" "main" {}
 ```
 
-### Activate Cockpit in a specific project
+### Manage Cockpit in a specific project
 
 ```terraform
 resource "scaleway_cockpit" "main" {
-  project_id = "11111111-1111-1111-1111-111111111111"
+project_id = "11111111-1111-1111-1111-111111111111"
```
Comment (yfodil) [inline docs/resources/cockpit.md:28]: Please fix indentation

File: docs/resources/cockpit.md [hunk 5]
```diff
@@ -15,17 +15,26 @@ For more information consult the [documentation](https://www.scaleway.com/en/doc
...
+}
+```
+
+### Choose a specific plan for Cockpit
+
+```terraform
+resource "scaleway_cockpit" "main" {
+project_id = "11111111-1111-1111-1111-111111111111"
+plan       = "premium"
```
Comment (yfodil) [inline docs/resources/cockpit.md:37]: Please fix indentation

File: docs/resources/cockpit.md [hunk 6]
```diff
@@ -35,43 +44,43 @@ resource "scaleway_cockpit" "main" {
 resource "scaleway_cockpit" "main" {}
 
 resource "scaleway_cockpit_grafana_user" "main" {
-  project_id = scaleway_cockpit.main.project_id
-  login      = "example"
-  role       = "editor"
+project_id = scaleway_cockpit.main.project_id
+login      = "example"
+role       = "editor"
```
Comment (yfodil) [inline docs/resources/cockpit.md:49]: Please fix indentation

File: docs/resources/cockpit.md [hunk 7]
```diff
@@ -35,43 +44,43 @@ resource "scaleway_cockpit" "main" {
...
-    - `alertmanager_url` - The alertmanager URL.
-    - `grafana_url` - The grafana URL.
-    - `traces_url` - The traces URL.
+- `plan_id` - (Deprecated) The ID of the current plan. Please use plan instead.
+- `endpoints` - (Deprecated) Endpoints. Please use scaleway_cockpit_source instead.
+- `metrics_url` - (Deprecated) The metrics URL.
+- `logs_url` - (Deprecated) The logs URL.
+- `alertmanager_url` - (Deprecated) The alertmanager URL.
+- `grafana_url` - (Deprecated) The grafana URL.
+- `traces_url` - (Deprecated) The traces URL.
```
Comment (yfodil) [inline docs/resources/cockpit.md:78]: Please fix indentation

File: docs/resources/cockpit.md [hunk 8]
```diff
@@ -35,43 +44,43 @@ resource "scaleway_cockpit" "main" {
...
+project_id = scaleway_cockpit.main.project_id
+login      = "example"
+role       = "editor"
 }
 
 provider "grafana" {
-  url  = scaleway_cockpit.main.endpoints.0.grafana_url
-  auth = "${scaleway_cockpit_grafana_user.main.login}:${scaleway_cockpit_grafana_user.main.password}"
+url  = scaleway_cockpit.main.endpoints.0.grafana_url
+auth = "${scaleway_cockpit_grafana_user.main.login}:${scaleway_cockpit_grafana_user.main.password}"
```
Comment (yfodil) [inline docs/resources/cockpit.md:54]: Please fix indentation

File: docs/resources/cockpit.md [hunk 9]
```diff
@@ -35,43 +44,43 @@ resource "scaleway_cockpit" "main" {
...
 }
 
 resource "grafana_folder" "test_folder" {
-  title = "Test Folder"
+title = "Test Folder"
```
Comment (yfodil) [inline docs/resources/cockpit.md:58]: Please fix indentation

**Review by yfodil — APPROVED** [review]

**Pull request merged**